lib/tickle/handler.rb

```diff
@@ -70,35 +71,39 @@ def guess_ordinal_and_unit
     def guess_special
       guess_special_other
       guess_special_beginning unless @next
-      guess_special_middle unless @next
-      guess_special_end unless @next
+      guess_special_middle    unless @next
+      guess_special_end       unless @next
     end
 
     private
 
     def guess_special_other
-      @next = @start.bump(:day, 2) if token_types.same?([:special, :day]) && token_of_type(:special).start == :other
-      @next = @start.bump(:week, 2)  if token_types.same?([:special, :week]) && token_of_type(:special).start == :other
+      @next = @start.bump(:day, 2)  if token_types.same?([:special, :day])  && token_of_type(:special).start == :other
+      @next = @start.bump(:week, 2) if token_types.same?([:special, :week]) && token_of_type(:special).start == :other
       @next = chronic_parse_with_start('2 months from now') if token_types.same?([:special, :month]) && token_of_type(:special).start == :other
-      @next = chronic_parse_with_start('2 years from now') if token_types.same?([:special, :year]) && token_of_type(:special).start == :other
+      @next = chronic_parse_with_start('2 years from now')  if token_types.same?([:special, :year])  && token_of_type(:special).start == :other
     end
 
     def guess_special_beginning
-      if token_types.same?([:special, :week]) && token_of_type(:special).start == :beginning then @next = chronic_parse_with_start('Sunday'); end
+      if token_types.same?([:special, :week])  && token_of_type(:special).start == :beginning then @next = chronic_parse_with_start('Sunday'); end
+      # this is busted
       if token_types.same?([:special, :month]) && token_of_type(:special).start == :beginning then @next = Date.civil(@start.year, @start.month + 1, 1); end
-      if token_types.same?([:special, :year]) && token_of_type(:special).start == :beginning then @next = Date.civil(@start.year+1, 1, 1); end
+      # ^^^^^^^^^^^^^^
```

Comment:
This has the same potential for error.  If the month is 12, and you add 1 to it, you get an invalid date... but it should be jumping to January instead.

I haven't fixed it because I haven't put a failing test through it yet.